libs/platform/ebpf_epoch.c

```diff
@@ -321,6 +321,9 @@ ebpf_epoch_enter()
         // Block until a thread entry is available.
         ebpf_semaphore_wait(_ebpf_epoch_cpu_table[current_cpu].epoch_table_semaphore);
         // After the semaphore is acquired, then there is at least 1 available epoch entry.
+
+        // Prevent APCs from running.
+        ebpf_enter_critical_region();
```

Comment:
should this be called before calling `ebpf_semaphore_wait`? otherwise there is still a time window between these 2 calls that the thread can be suspended?